cloudinit/distros/parsers/ifconfig.py

```diff
@@ -0,0 +1,252 @@
+# Copyright(C) 2022 FreeBSD Foundation
+#
+# Author: Mina Galić <me+FreeBSD@igalic.co>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import copy
+import re
+from functools import lru_cache
+from ipaddress import IPv4Address, IPv4Interface, IPv6Interface
+from typing import Dict, Optional, Tuple
+
+from cloudinit import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class Ifstate:
+    """
+    This class holds the parsed state of a BSD network interface.
+    It is itself side-effect free.
+    All methods with side-effects should be implemented on one of the
+    ``BSDNetworking`` classes.
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.inet = {}
+        self.inet6 = {}
+        self.up = False
+        self.options = []
+        self.nd6 = []
+        self.flags = []
+        self.mtu: int = 0
+        self.metric: int = 0
+        self.groups = []
+        self.description: Optional[str] = None
+        self.media: Optional[str] = None
+        self.status: Optional[str] = None
+        self.mac: Optional[str] = None
+        self.macs = []
+        self.vlan = {}
+        self.members = []
+
+    @property
+    def is_loopback(self) -> bool:
+        return "loopback" in self.flags or "lo" in self.groups
+
+    @property
+    def is_physical(self) -> bool:
+        # OpenBSD makes this very easy:
+        if "egress" in self.groups:
+            return True
+        if self.groups == [] and self.media and "Ethernet" in self.media:
+            return True
+        return False
+
+    @property
+    def is_bridge(self) -> bool:
+        return "bridge" in self.groups
+
+    @property
+    def is_bond(self) -> bool:
+        return "lagg" in self.groups
+
+    @property
+    def is_vlan(self) -> bool:
+        return "vlan" in self.groups or self.vlan
+
+    @property
+    def is_wlan(self) -> bool:
+        return "wlan" in self.groups or self.vlan
```

Comment:
Same question here.

Reply:
this shouldn't be checking for vlan at all
only wlan